clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractPartitionAssignor.java

```diff
@@ -58,8 +58,8 @@ public Subscription subscription(Set<String> topics) {
 
         Map<String, Integer> partitionsPerTopic = new HashMap<>();
         for (String topic : allSubscribedTopics) {
-            Integer numPartitions = metadata.partitionCountForTopic(topic);
-            if (numPartitions != null && numPartitions > 0)
+            int numPartitions = metadata.partitionCountForTopic(topic);
```

Comment:
Mentioned previously, but I think we had better retain the current behavior for `partitionCountForTopic`.